packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,244 +107,499 @@ type TileProps = {
+
+  return (
+    <Box className={cx('iui-thumbnail-icon', className)} ref={ref} {...rest}>
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileThumbnailPictureOwnProps>;
+
+TileThumbnailPicture.displayName = 'Tile.TileThumbnailPicture';
+
+// ----------------------------------------------------------------------------
+// Tile.QuickAction component
+
+const TileQuickAction = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-tile-thumbnail-quick-action', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+TileQuickAction.displayName = 'Tile.QuickAction';
+
+// ----------------------------------------------------------------------------
+// Tile.TypeIndicator component
+
+const TileTypeIndicator = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-tile-thumbnail-type-indicator', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+TileTypeIndicator.displayName = 'Tile.TypeIndicator';
+
+// ----------------------------------------------------------------------------
+// Tile.ButtonIcon component
+
+const TileButtonIcon = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      as='span'
+      className={cx('iui-button-icon', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'button'>;
+TileButtonIcon.displayName = 'Tile.ButtonIcon';
+
+// ----------------------------------------------------------------------------
+// Tile.Badge component
+
+const TileBadgeContainer = polymorphic.div(
+  'iui-tile-thumbnail-badge-container',
+);
+TileBadgeContainer.displayName = 'Tile.BadgeContainer';
+
+// ----------------------------------------------------------------------------
+// Tile.Name component
+type TileNameOwnProps = {
+  name?: string;
 };
 
+const TileName = React.forwardRef((props, ref) => {
+  const { className, children, name, ...rest } = props;
+  return (
+    <Box className={cx('iui-tile-name', className)} ref={ref} {...rest}>
+      {name ?? children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileNameOwnProps>;
+TileBadgeContainer.displayName = 'Tile.Name';
+
+// ----------------------------------------------------------------------------
+// Tile.NameIcon component
+
+const TileNameIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+  const { status, isLoading, isSelected, isNew } = useSafeContext(TileContext);
+
+  type TitleIconProps = {
+    isLoading?: boolean;
+    isSelected?: boolean;
+    isNew?: boolean;
+    status?: 'positive' | 'warning' | 'negative';
+  };
+
+  const TitleIcon = ({
```

Comment:
I'm getting this build error
```
src/core/Tile/Tile.tsx(281,15): error TS2604: JSX element type 'StatusIcon' does not have any construct or call signatures.
```

So doing this instead:
```
const TileNameIcon = React.forwardRef((props, ref) => {
  const { children, ...rest } = props;
  const { status, isLoading, isSelected, isNew } = useSafeContext(TileContext);

  const StatusIcon = !!status && StatusIconMap[status];
  let icon;

  if (isLoading) {
    icon = <ProgressRadial aria-hidden indeterminate />;
  }
  if (isSelected) {
    icon = <SvgCheckmark aria-hidden />;
  }
  if (isNew) {
    icon = <SvgNew aria-hidden />;
  }
  if (StatusIcon) {
    icon = <StatusIcon aria-hidden />;
  }

  return children || icon ? (
    <Box className='iui-tile-status-icon' ref={ref} {...rest}>
      {children ?? icon}
    </Box>
  ) : null;
}) as PolymorphicForwardRefComponent<'div'>;
TileNameIcon.displayName = 'Tile.NameIcon';
```
also returning empty if nothing is passed in so that there is no empty space/box in front of name when the subcomponent is present (especially on Storybook)

Reply:
StatusIcon should come before the other conditions

Reply:
Right, I moved it up.